src/Engine/ProtoCore/RuntimeStatus.cs

```diff
@@ -56,7 +57,7 @@ public struct WarningEntry
     public class RuntimeStatus
     {
         private ProtoCore.RuntimeCore runtimeCore;
-        private List<Runtime.WarningEntry> warnings;
+        private ConcurrentStack<Runtime.WarningEntry> warnings;
```

Comment:
ConcurrentStack<> seemed to be the closest thread safe collection implementation to List<>

Reply:
may be worth profiling this vs just a simple lock:
https://stackoverflow.com/questions/19326197/net-4-0-concurrent-collection-performance

I think it depends on the use case of course, also not sure this stack overflow question is the best source of benchmarks for this.

Reply:
Yeah..I was just thinking maybe a lock is enough...

Reply:
Although, If we always make a copy on the getter...it might be too slow